backend/handler/rooms.go

```diff
@@ -0,0 +1,158 @@
+package handler
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/jinzhu/copier"
+	"github.com/labstack/echo/v4"
+	"github.com/traP-jp/rucQ/backend/model"
+)
+
+func (s *Server) GetRooms(e echo.Context) error {
+	rooms, err := s.repo.GetRooms()
+
+	if err != nil {
+		e.Logger().Errorf("failed to get rooms: %v", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	var res []Room
+
+	if err := copier.Copy(&res, &rooms); err != nil {
+		e.Logger().Errorf("failed to copy models to response: %v", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	return e.JSON(http.StatusOK, res)
+}
+
+func (s *Server) PostRoom(e echo.Context, params PostRoomParams) error {
+	operator, err := s.repo.GetOrCreateUser(*params.XForwardedUser)
+
+	if err != nil {
+		e.Logger().Errorf("failed to get or create user: %v", err)
+
+		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
+	}
+
+	if !operator.IsStaff {
+		return echo.NewHTTPError(http.StatusForbidden, "Forbidden")
+	}
+
+	var req PostRoomJSONRequestBody
+
+	if err := e.Bind(&req); err != nil {
+		return e.JSON(http.StatusBadRequest, err)
+	}
+
+	var roomModel model.Room
```

Comment:
openapi.yamlの所にも少しコメントしましたが、部屋IDをパラメータとして受け取った上で既存の部屋の情報を持ってくる→その部屋の情報を書き換えて保存という流れになります。`repository`に個別の部屋情報を取得するための関数も追加する必要がありそうです
```suggestion
	roomModel, err := s.repo.GetRoomByID(uint(roomID))

	if err != nil {
		if errors.Is(err, model.ErrNotFound) {
			return echo.NewHTTPError(http.StatusNotFound, "Not found")
		}

		e.Logger().Errorf("failed to get room: %v", err)

		return echo.NewHTTPError(http.StatusInternalServerError, "Internal server error")
	}
```